Bundle chunks can exceed the configured split threshold

## Problem

`services.PartitionEntries` budgets a flat 200 bytes for the Bundle wrapper and never counts the comma that separates two entries in the serialized `entry` array. A chunk that holds many entries therefore grows past the threshold.

`internal/services/bundle_splitter.go:100-155`:

```go
const bundleOverheadBytes = 200
...
if len(currentPartition) > 0 && currentSize+entrySize+bundleOverheadBytes > thresholdBytes {
...
currentSize += entrySize   // the "," between two entries is never added
```

A chunk with N entries needs N-1 separator bytes that no one counts. As soon as N-1 grows past the leftover of the 200-byte budget, the chunk is larger than the threshold.

## Reproduction

A Bundle with 4000 tiny entries and a threshold of 4000 bytes produces a first chunk of **4411 bytes**. A Bundle with 20000 tiny entries and a threshold of 100000 bytes produces a first chunk of **114131 bytes**, about 14 percent over.

```
--- FAIL: FuzzSplitReassembleBundle/seed#5
    chunk 0 of 8 is 4411 bytes, above the threshold of 4000 bytes
--- FAIL: FuzzSplitReassembleBundle/seed#6
    chunk 0 of 2 is 114131 bytes, above the threshold of 100000 bytes
```

## Impact

The threshold is the payload limit of the DIMP server, set through `bundle_split_threshold_mb`. Splitting exists precisely to stay below that limit. A chunk above the limit is rejected by the server, so the pipeline step fails for the data that splitting was supposed to rescue. The drift grows with the number of entries, so a Bundle with many small resources is the most affected.

## Suggestion

Count the separator: add one byte per entry after the first while the partition grows. A test that asserts `len(json.Marshal(chunk)) <= threshold` for every chunk pins the contract.

Found by fuzzing (#646).

--- a/internal/services/bundle_splitter.go
+++ b/internal/services/bundle_splitter.go
@@ -141,15 +141,20 @@ func PartitionEntries(entries []map[string]any, thresholdBytes int) ([][]map[str
 			}
 		}
 
-		// Check if adding this entry would exceed threshold
-		if len(currentPartition) > 0 && currentSize+entrySize+bundleOverheadBytes > thresholdBytes {
+		// Check if adding this entry (plus the JSON separator preceding it
+		// in the serialized entry array) would exceed threshold
+		if len(currentPartition) > 0 && currentSize+1+entrySize+bundleOverheadBytes > thresholdBytes {
 			// Current chunk would exceed limit - start new chunk
 			partitions = append(partitions, currentPartition)
 			currentPartition = []map[string]any{}
 			currentSize = 0
 		}
 
-		// Add entry to current partition
+		// Add entry to current partition, counting the separator byte
+		// for every entry after the first
+		if len(currentPartition) > 0 {
+			currentSize++
+		}
 		currentPartition = append(currentPartition, entry)
 		currentSize += entrySize
 	}

--- a/tests/unit/bundle_splitter_test.go
+++ b/tests/unit/bundle_splitter_test.go
@@ -1,6 +1,8 @@
 package unit
 
 import (
+	"encoding/json"
+	"fmt"
 	"testing"
 	"time"
 
@@ -297,6 +299,70 @@ func TestPartitionEntriesGreedyAlgorithm(t *testing.T) {
 	// Verify all entries accounted for
 	assert.Equal(t, len(entries), totalEntries,
 		"All entries should be partitioned")
+}
+
+// TestSplitBundleChunksStayWithinThreshold verifies that every chunk produced by
+// SplitBundle serializes to at most the threshold, including the JSON separators
+// between entries, while still packing chunks close to the threshold.
+func TestSplitBundleChunksStayWithinThreshold(t *testing.T) {
+	testCases := []struct {
+		name           string
+		entryCount     int
+		thresholdBytes int
+	}{
+		{
+			name:           "many tiny entries with small threshold",
+			entryCount:     4000,
+			thresholdBytes: 4000,
+		},
+		{
+			name:           "many tiny entries with large threshold",
+			entryCount:     20000,
+			thresholdBytes: 100000,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			entries := make([]any, 0, tc.entryCount)
+			for i := 0; i < tc.entryCount; i++ {
+				entries = append(entries, map[string]any{
+					"resource": map[string]any{"id": fmt.Sprintf("%d", i)},
+				})
+			}
+			bundle := map[string]any{
+				"resourceType": "Bundle",
+				"id":           "tiny-entry-bundle",
+				"type":         "collection",
+				"entry":        entries,
+			}
+
+			result, err := services.SplitBundle(bundle, tc.thresholdBytes)
+			require.NoError(t, err)
+			require.True(t, result.WasSplit)
+
+			totalEntries := 0
+			for i, chunk := range result.Chunks {
+				totalEntries += len(chunk.Entries)
+
+				serialized, err := json.Marshal(models.ConvertChunkToBundle(chunk))
+				require.NoError(t, err)
+				assert.LessOrEqual(t, len(serialized), tc.thresholdBytes,
+					"chunk %d serializes to %d bytes, exceeding threshold %d",
+					i, len(serialized), tc.thresholdBytes)
+
+				// All chunks except the last must stay close to the threshold,
+				// otherwise splitting degrades into too many small requests
+				if i < len(result.Chunks)-1 {
+					assert.GreaterOrEqual(t, len(serialized), tc.thresholdBytes*9/10,
+						"chunk %d serializes to %d bytes, far below threshold %d",
+						i, len(serialized), tc.thresholdBytes)
+				}
+			}
+			assert.Equal(t, tc.entryCount, totalEntries,
+				"all entries should be preserved across chunks")
+		})
+	}
 }
 
 // TestBundleMetadataRoundTrip verifies metadata survives extraction and restoration

